app/models/vm_or_template.rb

```diff
@@ -1673,9 +1673,8 @@ def service
     direct_service.try(:root_service)
   end
 
-  def raise_is_available_now_error_message(request_type)
-    msg = send("validate_#{request_type}")[:message]
-    raise MiqException::MiqVmError, msg unless msg.nil?
+  def raise_not_supported_error_message(request_type)
+    raise MiqException::MiqVmError, unsupported_reason(request_type) unless send("supports_#{request_type}?")
```

Comment:
this can be `unless supports?(request_type)` without the `send`
and TBH I would inline this method, it's just one line, has no abstraction use and move the exception handling closer to the methods.